### ISSUE TITLE
Not updating switch icon for async switch (mqtt)

### ISSUE THREAD

**Issue by micw** [openhab/openhab#4833]
The following problem occurs on the android app but only when using OpenHAB 1 as backend. With OpenHAB 2 it works. So I assume it's a backend related issue, so I post it here:

I have several mqtt-bound switches. They have only 2 states (on/off). Communication is asynchronous. An "on" command is sent to the switch as mqtt message, after the switch has switched, it responses with another mqtt message that indicates the new "on" state.

I use the android app.
When I use OpenHAB 1 as backend, there is a small "lamp" icon in front of the switch item. When I tap the switch, the lamp switches on/off but the lamp icon do not change. If i reload the view, the lamp icon shows the correct state.

When I use OpenHAB 2 as backend, there is a small "power switch" icon in front of the switch item. When I tap the switch, the lamp switches on/off and the power switch icon changes to show the correct lamp state.

So something seems to be wrong on OH1 that works on OH2.

**Comment by micw** [comment]
Update: Today I put Nginx between the backend and the app for SSL termination. This showed me an interesting detail:

```
80.153.43.250 - hab [28/Nov/2016:07:07:26 +0100] "POST /rest/items/Garten_licht HTTP/1.1" 201 0 "-" "-"

2016/11/28 07:07:47 [error] 13270#0: *58 upstream timed out (110: Connection timed out) while reading response header from upstream, upstream: "http://192.168.1.5:8080/rest/sitemaps/default/demo"

80.153.43.250 - hab [28/Nov/2016:07:07:47 +0100] "GET /rest/sitemaps/default/demo HTTP/1.1" 504 182 "-" "-"
80.153.43.250 - hab [28/Nov/2016:07:07:47 +0100] "GET /rest/sitemaps/default/demo HTTP/1.1" 200 1689 "-" "-"
```

It seems that my OH1 instance hangs on the get-sitemap request if it's fired at the same time as the POST to the switch. Nginx times out the proxy connection after 30s by default - so with nginx I get status/icon update after 30s. Without nginx there's no such timeout and the UI fails to update the status.

**Comment by micw** [comment]
I found out that the proxy error is just because the long polling. So it's OK if there is no event that causes the long-polling request to return.

The problem is, that the state change of the switch does not trigger the long-polling request to return. I have tested it with curl:

curl http://192.168.1.5:8080/rest/sitemaps/default/demo -H "X-Atmosphere-Framework: 1.0" -H "X-Atmosphere-Transport: long-polling"

This waits until something changes. Now I change the state of the switch from ON to OFF. The curl is still waiting. When I do the same, on OH2, it works and curl returns with new state.
